### ISSUE TITLE
Action not allowed when logout

### ISSUE THREAD

**Issue by red-ruby-quay** [derricksmith/phpsaml#129]
Thanks for your plugins, It works for me, 

This is due to SSO Logout. I'm using keycloak as idp, and using idp logout parameter as the link for SLO request.
The problem is, when I logout the account that successfully logged in, the redirect goes to {{url_base}}/plugins/phpsaml/front/slo.php and displayed this error page. But the post binding for the logout seems worked

![saml_error_redirect](https://user-images.githubusercontent.com/105847027/222326577-3038069a-8a75-40bb-822b-432237a87181.png)

I think the cookies and the session has been destroyed while the route redirected to .../front/slo.php, so the access to that route is not allowed.

Could you check the logic flow and fix that error? So when the user logout, the session and cookies expired and only redirected to the login page again, not to the slo file.

Btw, I had change the url of SSO on phpsaml.class.php (line 88) so it could redirect to base url or login page. But nothing happened...

**Comment by DonutsNL** [comment]
Im not sure SLO is used. It simply performs a die() after loading the glpi includes. As far as im aware, the file has never been hit in any of the setups i know.

Please try to redirect the user to the glpi index page instead.

**Comment by red-ruby-quay** [comment]
No, I've use SLO as the logout request. But as you could see, still redirected to the SLO link itself even after the cookies and sessions deleted. 

<img width="480" alt="error_redirect" src="https://user-images.githubusercontent.com/105847027/224893829-0925d0e9-7bb8-43e8-bdbc-eaee580c8fcb.png">

and what I knew before, there is no die() performed after loading glpi includes in phpsaml.class.php
<img width="430" alt="error_redirect_2" src="https://user-images.githubusercontent.com/105847027/224893847-1f851078-a0a7-4952-9dc3-730c17692d24.png">

I've tried to change the redirect link (in picture 1), but nothing happened, still error redirect even the request status is OK 200. 

Could you tell me what to do and what file that I must change the code to take the correct redirect result?

**Comment by DonutsNL** [comment]
Im not sure what i am looking at in the first screenshot. I do recognize the second screenshot. Looking at the current master branch (of our code), the slo is performing a die(); basically not doing anything atm. As you can see for yourself.

![image](https://user-images.githubusercontent.com/97617761/235319860-61079f06-2378-4254-ae69-278b80c18c57.png)

What version of phpsaml are you using currently, and can you share the slo.php with me?

**Comment by red-ruby-quay** [comment]
I'm sorry for the super late respons...
I use the latest version of this project (tag 1.2.1, master branch)

This is the slo.php 
<img width="798" alt="slo php" src="https://github.com/derricksmith/phpsaml/assets/105847027/06c6735e-f02c-41b7-892d-4a9355e97659">

**Comment by red-ruby-quay** [comment]
> I'm sorry for the super late respons... I use the latest version of this project (tag 1.2.1, master branch)
> 
> This is the slo.php <img alt="slo php" width="798" src="https://user-images.githubusercontent.com/105847027/267201656-06c6735e-f02c-41b7-892d-4a9355e97659.png">

What do I do? I mean like delete the ```die();``` code and just redirect it to login page ```auth()->login();```?

**Comment by red-ruby-quay** [comment]
Thanks for the advice.. I've managed to get in how to logout without stuck in {{server_url}}/plugins/phpsaml/front/slo.php "not allowed method".

Here's my code on slo.php (replacing all codes)

```
<?php
 /* Redirect browser: change the location based on latest server deployment place, no need to change 'true' value for default header replacement and default response code (302)*/ 
header("Location: {{your_base_url}}"); 
/* Make sure that code below does not get executed when we redirect. */
exit;
?>
```
It is pretty simple, but you could modified it anything you want...

I've tried @ChrisGralike  advice that "```21``` and ```22```" rules must be used. But it couldn't, because the session has already been cleared while the code try to use some of onelogin saml library for another authentication process again. Instead you could use this code => plugins/phpsaml/front/ssoRequest.php but with some modifications (redirect to SLO then to base url instead of login redirect).


For anyone that couldn't know how to fix this slo bug yet (or maybe you could call it undone codes page...), try my advice in this comment. Make sure to share your accomplishment in this comment issue section for knowledge sharing and better improvement.

Anyway, thank you very much @ChrisGralike @DonutsNL !!!
